### DISCUSSION TITLE
How to query results only when the search input is not empty?

### DISCUSSION THREAD

**Discussion by jpomykala** [algolia/instantsearch#5317] [Q&A] [answered]
I would like to prevent from sending a query request when the search input is empty. How to configure it? Currently, the only way which I found is to not render `<SearchBox />` component.

**Comment by francoischalifour** [comment, marked as answer]
There's no boolean configuration for this, but you can implement this at the search client level:

```js
const originalSearchClient = algoliasearch(
  "latency",
  "6be0576ff61c053d5f9a3225e2a90f76"
);
const searchClient = {
  ...originalSearchClient,
  search(requests) {
    if (requests.every(({ params }) => !params.query)) {
      return Promise.resolve({
        results: requests.map(() => ({
          hits: [],
          nbHits: 0,
          nbPages: 0,
          page: 0,
          processingTimeMS: 0
        }))
      });
    }

    return originalSearchClient.search(requests);
  }
};
```

I created a [sandbox](https://codesandbox.io/s/billowing-meadow-usjovp?file=/App.tsx:250-742) to show you.

We cover this in the ["Detecting empty search requests"](https://www.algolia.com/doc/guides/building-search-ui/going-further/conditional-requests/js/#detecting-empty-search-requests) section of our documentation (soon available for React InstantSearch Hooks).
